internal/ownerhandling/annotation.go

```diff
@@ -81,15 +90,43 @@ func (s *OwnerStrategyAnnotation) SetControllerReference(owner, obj metav1.Objec
 }
 
 func (s *OwnerStrategyAnnotation) IsOwner(owner, obj metav1.Object) bool {
-	ownerRefs := s.getOwnerReferences(obj)
-	for _, ownerRef := range ownerRefs {
-		if ownerRef.UID == owner.GetUID() {
+	ownerRefComp := s.ownerRefForCompare(owner)
+	for _, ownerRef := range s.getOwnerReferences(obj) {
+		if s.referSameObject(ownerRefComp, ownerRef) {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *OwnerStrategyAnnotation) IsController(
+	owner, obj metav1.Object) bool {
+	ownerRefComp := s.ownerRefForCompare(owner)
+	for _, ownerRef := range s.getOwnerReferences(obj) {
+		if s.referSameObject(ownerRefComp, ownerRef) &&
+			ownerRef.Controller != nil &&
+			*ownerRef.Controller {
 			return true
 		}
 	}
 	return false
 }
 
+func (s *OwnerStrategyAnnotation) RemoveOwner(owner, obj metav1.Object) {
+	ownerRefs := s.getOwnerReferences(obj)
+	foundIndex := -1
+	for i, ownerRef := range ownerRefs {
+		if owner.GetUID() == ownerRef.UID {
```

Comment:
Some places the comparison is made with `referSameObject` and some places by comparing the UID. Is there are reason to have two different approaches?